### DISCUSSION TITLE
Selecting Simple Window Switcher prevents desktop slideshow from working after login

### DISCUSSION THREAD

**Discussion by UCyborg** [valinet/ExplorerPatcher#895] [General]
But it starts working if you go to Settings app and re-set the desired interval for changing desktop background. Tested on Win11 version 22000.493 and ExplorerPatcher version 22000.469.41.16. Other tweaking programs have been temporarily disabled to rule out possibility of the issue occurring in the combination with one of those.

**Comment by valinet** [comment]
I have just checked that, I cannot reproduce. OS build 22000.493 and EP latest pre-release, although regarding sws not much has changed in the latest pre-release compared to the latest release.

**Comment by UCyborg** [comment]
Strange, I've no clue what would cause it. I figured just visiting the background section in Settings app is enough to get it working again, but why does Simple Window Switcher manage to break it in the first place here...

**Reply by valinet** [reply]
sws sens an undocumented message to the Progman window in order to have it create the wallpaper window so that subsequently sws is able to display an image of the desktop wallpaper in the switcher.

**Reply by valinet** [reply]
If there’s a problem, that may be the source of it. Too bad I can’t repro what you describe. This area of sws ate A TON of work hours for something relatively unimpressing in the end because things are pretty convoluted. Maybe I have to revisit it, despite passionately hating the subject.

When you say the slideshow is broken, does right clicking on the desktop still show the “Next background image” entry?

**Reply by UCyborg** [reply]
Right, I should've clarified, by broken I meant simply that the background doesn't change at set interval, but it can be rolled manually by "Next background image" option. I also noticed that notifications about inserted media in optical drive or inserted USB flash drive also get it going again.

Yesterday when I booted the virtual machine once it was working by lucky coincidence without me doing anything, but after logging off and logging in the slideshow was again on halt.

It may be of relevance that I'm testing on underpowered virtual machine. Host has AMD Phenom II X4 920 CPU clocked at 3 GHz (slight overclock from 2,8 GHz, 'been stable for many years), 4 GB of RAM and NVIDIA GeForce GTX 750 Ti. Virtual machine has 4 virtual CPUs and merely 1,5 GB of RAM, can't really sacrifice more, but Windows and simpler programs work. Virtualization environment is older VMware Workstation 15.5.7 (which doesn't officially support Windows 11) running on Windows 10 20H2 (precisely 10.0.19042.508).

Maybe the key to reproduction is slower operating environment. I wonder if it would occur here running on bare metal, not sure when I'll get around to installing it that way. Maybe you can try slowing your machine down, eg. using power saver power plan, [booting with less CPU cores](https://helgeklein.com/blog/how-to-limit-cpu-ram-via-the-windows-boot-configuration/).

**Reply by valinet** [reply]
The question is whether this is a bug in Windows 10 or regular Windows as well. In that case, it's not really related to Simple Window Switcher or ExplorerPatcher. For example, the bug where the "Customize system tray icons" Control Panel window would crash when the minutes change on the system clock happens in Windows 10 as well, without ExplorerPatcher installed, so yeah, Windows has quite a few bugs, actually.

That the slideshow is paused when the computer has few resources would actually make sense, idk...

**Comment by UCyborg** [comment]
It's like something doesn't get triggered under certain specific circumstance when selecting SWS. AFAIK, there's only a setting in power settings to select whether slideshow works or is paused, I doubt MS made it pause intentionally in other conditions. Toggling that setting also gets it going again.

I tried hiding desktop icons if it would change anything, it didn't. It also doesn't seem to matter if Explorer is started during login process or restarted at a later point, when I presume there's less going on in the background.

Regarding the virtual machine, particularly its 1,5 GB of RAM, Win11 alone actually does fit quite nicely in a little over 1 GB of memory.

![Win11 screenshot](https://i.imgur.com/ejIl33G.png)

If I only have Process Hacker open, it reports RAM usage at about 1,04 GB.
